### PR TITLE
UCP/RNDV: Add infrastructure for rndv pipeline

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Preparations for RNDV-pipeline protocol

### CODE REVIEW

File: src/ucp/core/ucp_request.h [hunk 1]
```diff
@@ -231,14 +234,25 @@ struct ucp_request {
                             uint8_t        rkey_index[UCP_MAX_LANES];
                         };
 
-                        /* Used by rndv/put */
+                        /* Used by version 2 rendezvous protocols */
```
Comment (dmitrygx) [inline src/ucp/core/ucp_request.h:237]: `... version 2 of rendezvous protocols`?

File: src/ucp/core/ucp_request.h [hunk 2]
```diff
@@ -231,14 +234,25 @@ struct ucp_request {
...
-
-                            /* which lanes need to send atp */
-                            ucp_lane_map_t atp_map;
-                        } put;
+                            /* Data start offset of this request */
+                            size_t offset;
+
+                            /* Used by rndv/put and rndv/put/frag */
+                            struct {
+                                /* which lanes need to flush (0 in fence mode) */
```
Comment (dmitrygx) [inline src/ucp/core/ucp_request.h:244]: start from capital letter

File: src/ucp/core/ucp_request.h [hunk 3]
```diff
@@ -231,14 +234,25 @@ struct ucp_request {
...
+                                ucp_lane_map_t flush_map;
+
+                                /* which lanes need to send atp */
```
Comment (dmitrygx) [inline src/ucp/core/ucp_request.h:247]: start from capital letter

File: src/ucp/rndv/proto_rndv.inl [hunk 4]
```diff
@@ -133,14 +133,76 @@ ucp_proto_rndv_rkey_destroy(ucp_request_t *req)
...
+ * including when the request is a fragment and starts from nonzero offset.
+ */
+static UCS_F_ALWAYS_INLINE size_t
+ucp_proto_rndv_bulk_max_payload(ucp_request_t *req,
+                                const ucp_proto_rndv_bulk_priv_t *rpriv,
+                                const ucp_proto_multi_lane_priv_t *lpriv)
+{
+    size_t total_length = ucp_proto_rndv_request_total_length(req);
+    size_t max_frag_sum = rpriv->mpriv.max_frag_sum;
+    size_t offset = req->send.rndv.offset + req->send.state.dt_iter.offset;
```
Comment (dmitrygx) [inline src/ucp/rndv/proto_rndv.inl:172]: unaligned by `=`

File: src/ucp/rndv/proto_rndv.c [hunk 5]
```diff
@@ -297,19 +297,23 @@ ucs_status_t ucp_proto_rndv_rts_init(const ucp_proto_init_params_t *init_params)
...
+        .super.min_frag_offs = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.max_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.am.max_bcopy),
+        .super.max_iov_offs  = UCP_PROTO_COMMON_OFFSET_INVALID,
+        .super.hdr_size      = 0,
+        .super.memtype_op    = UCT_EP_OP_LAST,
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RESPONSE,
+        .remote_op_id        = UCP_OP_ID_RNDV_RECV,
+        .perf_bias           = context->config.ext.rndv_perf_diff / 100.0,
+        .mem_info.type       = init_params->select_param->mem_type,
+        .mem_info.sys_dev    = init_params->select_param->sys_dev,
```
Comment (dmitrygx) [inline src/ucp/rndv/proto_rndv.c:316]: no need for trailing comma

**Review by dmitrygx — APPROVED** [review]
besides minor comments

**Review by dmitrygx — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**